mm2src/coins/lp_coins.rs

```diff
@@ -2167,21 +2204,44 @@ impl CoinsContext {
         tokens: Vec<MmCoinEnum>,
     ) -> Result<(), MmError<PlatformIsAlreadyActivatedErr>> {
         let mut coins = self.coins.lock().await;
+        let mut platform_coin_tokens = self.platform_coin_tokens.lock();
+
         if coins.contains_key(platform.ticker()) {
             return MmError::err(PlatformIsAlreadyActivatedErr {
                 ticker: platform.ticker().into(),
             });
         }
 
+        let platform_ticker = platform.ticker().to_string();
         coins.insert(platform.ticker().into(), platform);
 
         // Tokens can't be activated without platform coin so we can safely insert them without checking prior existence
         for token in tokens {
-            coins.insert(token.ticker().into(), token);
+            coins.insert(token.ticker().into(), token.clone());
+            platform_coin_tokens
+                .entry(platform_ticker.clone())
+                .or_default()
+                .insert(0, token.ticker().to_string());
         }
+
         Ok(())
     }
 
+    /// Get enabled coins to disable.
+    pub async fn get_coins_to_disable(&self, platform_ticker: &str, ticker: &str) -> Vec<String> {
+        let coins = self.platform_coin_tokens.lock();
+        let mut coins_storage = vec![];
+
+        if ticker == platform_ticker {
+            coins_storage.extend(coins.get(platform_ticker).unwrap_or(&vec![]).clone());
+        };
+
+        if !coins_storage.contains(&ticker.to_owned()) {
+            coins_storage.push(ticker.to_string());
+        }
+        coins_storage
+    }
```

Comment:
I think this function can be simplified as follows
``` rust
pub async fn get_coins_to_disable(&self, ticker: &str) -> Vec<String> {
        let coins = self.platform_coin_tokens.lock();
        let mut coins_storage = coins.get(ticker).unwrap_or(&vec![]).clone();
        coins_storage.push(ticker.to_string());
        coins_storage
    }
```

Reply:
ok thanks